### PR TITLE
Use lzma_sdk in 7zip recipe

### PR DESCRIPTION
**Pull request by danimtb**
Specify library name and version:  **7zip/19.00**

Related to #251 

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

cc/ @madebr

### CODE REVIEW

**Comment by danimtb** [comment]
@uilianries we need to relax the check in the hook

**Comment by madebr** [comment]
@danimtb I hope you'll also check 7zip using mingw?
Then #227 can be closed

**Comment by danimtb** [comment]
Didn't check it yet with this new PR

File: recipes/7zip/19.00/conanfile.py [hunk 1]
```diff
@@ -10,36 +9,27 @@ class Package7Zip(ConanFile):
...
     _msvc_platforms = {
         'x86_64': 'x64',
         'x86': 'x86',
     }
 
+    def source(self):
+        tools.download(self.conan_data["sources"][self.version]["url"], "7zip.7z")
+        tools.check_sha256("7zip.7z", self.conan_data["sources"][self.version]["sha256"])
+        self.run("7zr x 7zip.7z")
+
```
Comment (madebr) [inline recipes/7zip/19.00/conanfile.py]:
```suggestion
    os.unlink("7zip.7z")

```

Comment (madebr) [reply recipes/7zip/19.00/conanfile.py]: Sorry for the wrong indenting :D

**Comment by danimtb** [comment]
Closing this in favor of the approach discussed in #414

**Pull request closed without merging**